### ISSUE TITLE
Points to take care off while preparing a data-set for training

### ISSUE THREAD

**Issue by Bahramudin** [tensorflow/models#6208]
### System information
- **What is the top-level directory of the model you are using**: Object Detection 
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: N/A
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Windows 64 Bit
- **TensorFlow installed from (source or binary)**: Binary
- **TensorFlow version (use command below)**: 1.12.0
- **Bazel version (if compiling from source)**: N/A
- **CUDA/cuDNN version**: 9.0 / 7.2.5
- **GPU model and memory**:GTX 1080 8GB 
- **Exact command to reproduce**: N/A

### Describe the problem
Please don't mind why I am asking this question. Because I want to possibly improve the accuracy of the Object Detecting. Because training is done on a data-set, so part of the accuracy depends on how the quality of our data-set is. I think a good quality of dataset will help to improve the detecting accuracy. 

So as an expert in AI and object detection, what is the important point to take care of, when collecting the images, labeling and so on.  What kind of images is better for training. And while labeling, how to label the images. Labeling is easy, but the points to take care off is very important, such as the below image:
![image](https://user-images.githubusercontent.com/13574895/52844789-5da0d480-3140-11e9-9b31-653a68bc3ced.png)
 And so on points.

**The second** part of the question is for what kind of object we need at least how much images to collect and label. For example, training **small objects** vs training **big objects**? And how much image is very enough for both kind. And also how long we need to train the dataset for each kind of data.

These points I think are very important to know, maybe there is no problem with Framework, but the problem with data, or no problem with data, but the problem with parameters to set and use during each kind data... and so on.

So please share the ideas and the experience if you have or know to help others may have the same problem or question about training as well.

Thanks, everyone!

**Comment by auslaner** [comment]
First off, I'm most certainly not an expert on AI and object detection. But I do have some thoughts and a little bit of experience.

I think the answer to all of these questions is going to depend on your training data and the goal of your particular project. As an example, "looser" bounding boxes could provide the algorithm with helpful background information that could help improve detection accuracy. However, if you train your model using only "fixed" data, that might enable you to detect collections between objects since the bounding boxes would more closely represent the edges of the objects being detected.

My suggestion would be to label everything with "fixed" bounding boxes and then use a simple script to generate "loose" and "very fixed" bounding boxes from those. Then you could train all three models and see what works best for your particular problem. I would be surprised if what works best for one application would also be universally best for all other object detection applications.

Most of these same points apply to the question of big and small objects. You would need to evaluate your model to see if it performs better or worse on a particular size of object and provide more training data for the size class with poorer performance. That said, it stands to reason that small objects probably need more training data than large objects since those objects by definition have fewer pixels for the algorithm to learn about.

**Comment by Bahramudin** [comment]
@auslaner Thanks for the reply! I have another question I want to ask, which I am wondering why need to provide such a huge amount of data for training the network? For example, at current dataset almost all the data are over 95% similar whit each other, so it means that if summarize the whole dataset, then it will be only a few images left at the end because of the high similarity whit each other. 

So why we cannot provide images of only the most important part of the object to train instead of such a huge amount?  For example, for the above bottle of Cola, we say to the network that this object from the front looks like this, and from the top like this, and from the left like this, and from the right like this, and enough. So this can prevent from wasting a lot of time collecting only similar images which are a very waste of time and sources.

**Comment by auslaner** [comment]
I wouldn't say you _need_ a huge amount of data. Imagine if you were trying to train a model to recognize green images and red images:

![image](https://user-images.githubusercontent.com/44983332/54141105-a586e480-43ea-11e9-9fac-7eea563f1940.png) ![image](https://user-images.githubusercontent.com/44983332/54141251-ebdc4380-43ea-11e9-8152-607af235b7a1.png)

You only need a small handful of training images if you know that the input images will be entirely either red or green. But now lets say you want to classify this image using our model:

![image](https://user-images.githubusercontent.com/44983332/54142297-00214000-43ed-11e9-9dea-3ef9b882600d.png)

Maybe we want our model to classify such as image as red since most pixels are red. Or maybe we want the model to classify images as green if any pixel contains green. Either way, we would need to train the model with more training data in order to make achieve the result we want. 

All of this is to say, that the amount of training data we need is going to vary based on how variable our input data is going to be. Generally, the more training data we provide, the better the model will be able to generalize to new input data. As you said though, if your training data is 95% similar to existing training data, it won't be as helpful to learning.

**Comment by Bahramudin** [comment]
@auslaner Hmmm, Yes I got it. But if be like that, then the very different of the images will cause that they seem two or more different objects (not the same object), but if you keep all they look like the same object, then the similarity will go over 95%. 

Now in the image above, while the network got the input like that, then it must calculate the probability, for example, the above image 90% is Red and 10% is Green, **SO IT CALL AI**.

So if I must also tell the network the image is either red or green, then it means I hardcoded, Or it means it will just be able to detect the object which looks like the same as data set. If be like this, then **IT IS NOT AI**.

So, in my opinion, the AI in the current state is something toy programming, some just the kids play some game, nothing else.

The AI must be like I only tell you only one image of the object, then rest of detection you detect by yourself, it that case it will become useful. But now NO.

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be an old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.
